src/dashboard/src/components/filesystem_ajax/views.py

```diff
@@ -331,8 +331,19 @@ def create_arranged_sip(staging_sip_path, files, sip_uuid):
     # Create SIP object
     sip_name = staging_sip_path.split('/')[1]
     sip_path = os.path.join(shared_dir, 'watchedDirectories', 'SIPCreation', 'SIPsUnderConstruction', sip_name)
+    currentpath = sip_path.replace(shared_dir, '%sharedPath%', 1) + '/'
     sip_path = helpers.pad_destination_filepath_if_it_already_exists(sip_path)
-    sip_uuid = databaseFunctions.createSIP(sip_path.replace(shared_dir, '%sharedPath%', 1)+'/', sip_uuid)
+    sips = models.SIP.objects.filter(uuid=sip_uuid)
+    # Update the already-created SIP with its path
+    if sips.count():
+        sip = sips[0]
+        if sip.currentpath is not None:
+            return "Provided SIP UUID ({}) belongs to an already-started SIP!".format(sip_uuid)
+        sip.currentpath = currentpath
+        sip.save()
+    # Create a SIP object if none exists
+    else:
+        databaseFunctions.createSIP(currentpath, sip_uuid)
```

Comment:
This seems like it would be easier with a `.get` and try/catch? eg

```
try:
    sip = models.SIP.objects.get(uuid=sip_uuid)
    if sip.currentpath is not None:
        return "Provided SIP UUID ({}) belongs to an already-started SIP!".format(sip_uuid)
    sip.currentpath = currentpath
    sip.save()
except models.SIP.DNE:
    databaseFunctions.createSIP(currentpath, sip_uuid)
```